src/log-pcap-stream.c

```diff
@@ -0,0 +1,309 @@
+      */
+      while (tags != NULL) {
+          DetectTagDataEntry *current_tag = tags;
+          tags = tags->next;
+          /* Initialize the tag's output(s) PCAP stream if not already done. */
+          if (current_tag->pcap_file == NULL) {
+              /**
+               * Find the Signature instance inside the packet that matches
+               * the tag's SID.
+               */
+               const PacketAlert *tagAlert = NULL;
+               for (uint16_t x=0; x < p->alerts.cnt; x++) {
+                   if (p->alerts.alerts[x].s->id == current_tag->sid) {
+                       tagAlert = &p->alerts.alerts[x];
+                       break;
+                   }
+               }
+               if (tagAlert == NULL) {
+                   /**
+                    * This case happens when a rule hits its threshold
+                    * settings. The alert doesn't get generated but the
+                    * tagging will still occur. Skip the tag.
+                    */
+                    continue;
+               }
+               current_tag->pcap_file = SetupTaggedPcap(p, tagAlert->s,
+                       tv->id, td->count);
+
+               if (td->stream_output_ctx->session_dump_enabled) {
+                   TcpSession *session = (TcpSession *) p->flow->protoctx;
+                   if (session != NULL) {
+                       LogTcpSession(session,
+                               current_tag->pcap_file->pcap_dumper,
+                               p);
+                   }
+               }
+          }
+          DumpTaggedPacket(current_tag->pcap_file->pcap_dumper, p);
+      }
+}
+
+/**
+ *  \brief StreamPcapLogRegister Register the logger to the output-packet
+ *   root logger.
+ */
+void StreamPcapLogRegister(void)
+{
+    SCLogNotice("StreamPcapLogRegister Enter");
+    OutputRegisterPacketModule(LOGGER_PCAP,        // Logger ID
+            MODULE_NAME,                           // Logger name
+            "stream-pcap-log",                     // Configuration name
+            StreamPcapLogInitCtx,                  // Output init function
+            StreamPcapLog,                         // Packet logger function
+            StreamPcapLogCondition,                // Packet condition function
+            StreamPcapLogThreadInit,               // Thread init function
+            StreamPcapLogThreadDeInit,             // Thread deinit
+            NULL);                                 // Thread print stats
+}
+
+/**
+ * \brief Determines whether or not to log this packet.
+ * \param tv thread-specific variables.
+ * \param p Pointer to current packet being processed.
+ * \return TRUE if the packet should be logged.
+ * \return FALSE if we do not need to log the packet.
+ */
+static int StreamPcapLogCondition(ThreadVars *tv, const Packet *p)
+{
+    /* Flow is necessary for tag lookups. Reject invalid packets. */
+    if ((p->flags & PKT_HAS_FLOW && !(p->flags & PKT_IS_INVALID)) ||
```

Comment:
This can be a single check: `(p->flags & (PKT_HAS_FLOW|PKT_IS_INVALID)) == PKT_HAS_FLOW)`

Not sure if the compiler would produce the same code, so net effect might be 0.